src/components/Auth/AuthForm.tsx

```diff
@@ -0,0 +1,159 @@
+import classNames from 'classnames';
+import React, { useEffect, useState } from 'react';
+import { createUser, getUserByEmail } from '../../api/users';
+import { User } from '../../types/User';
+
+export type Props = {
+  onLogin: (user: User) => void,
+};
+
+export const AuthForm: React.FC<Props> = ({ onLogin }) => {
+  const [email, setEmail] = useState('');
+  const [name, setName] = useState('');
+  const [needToRegister, setNeedToRegister] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const saveUser = (user: User) => {
+    localStorage.setItem('user', JSON.stringify(user));
+    onLogin(user);
+  };
+
+  useEffect(() => {
+    const userData = localStorage.getItem('user');
+
+    if (!userData) {
+      return;
+    }
+
+    try {
+      const user = JSON.parse(userData) as User;
+
+      onLogin(user);
+    } catch (error) {
+      // Need to login
```

Comment:
Remove comment and do logic when catch error. If you have no actions, then it is better not to catch her, let her float higher. Otherwise, you hide it - if you caught it and did not process it, no one will see it anymore (

Reply:
Ok, I got it!